File: RpnCalculator/src/rpn/operator/AbsoluteOperator.java
package rpn.operator;

import rpn.RpnStack;

public class AbsoluteOperator implements Operator {

	@Override
	public int doOperation(RpnStack numbers) {
		int n = numbers.pop();
		int result = getAbsolute(n);
		numbers.push(result);
		return result;
	}

	private int getAbsolute(int n) {
		int result;
		if (n > 0)
			result = n;
		else if (n < 0)
			result = -1 * n;
		else
			result = n;
		return result;
	}

	@Override
	public boolean handlesOperatorCharacter(String operand) {
		return operand.contentEquals("||");
	}

}

File: RpnCalculator-Scala/src/rpn/scala/operators/AbsoluteOperator.scala
package rpn.scala.operators

import rpn.scala.RpnStack

class AbsoluteOperator extends Operator {
  val ABSOLUTE = "||"

  def performOperation(numbers: RpnStack): Int = {
    val n = numbers.pop();
    val result = {
      if (isPositive(n))
        n
      else
        changeSign(n)
    }
    numbers.push(result);
    return result
  }

  def handlesOperatorCharacter(operand: String): Boolean = {
    operand == ABSOLUTE;
  }

  def changeSign(n: Int) = { -1 * n }

  private def isPositive(n: Int): Boolean = {
    n > 0
  }

}